lib/usdUfe/ufe/UsdUndoSetDefaultPrimCommand.cpp

```diff
@@ -0,0 +1,60 @@
+//
+// Copyright 2023 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "UsdUndoSetDefaultPrimCommand.h"
+
+#include <usdUfe/base/tokens.h>
+#include <usdUfe/ufe/Utils.h>
+#include <usdUfe/undo/UsdUndoBlock.h>
+#include <usdUfe/utils/editRouter.h>
+#include <usdUfe/utils/editRouterContext.h>
+
+#include <pxr/usd/usdGeom/imageable.h>
+
+namespace USDUFE_NS_DEF {
+
+UsdUndoSetDefaultPrimCommand::UsdUndoSetDefaultPrimCommand(const UsdPrim& prim)
+    : Ufe::UndoableCommand()
+    , _prim(prim)
+{
+    _stage = _prim.GetStage();
+    if (!_stage) {
+        return;
+    }
+}
+
+UsdUndoSetDefaultPrimCommand::~UsdUndoSetDefaultPrimCommand() { }
+
+void UsdUndoSetDefaultPrimCommand::execute()
+{
+    UsdUndoBlock undoBlock(&_undoableItem);
+
+    // Check if the layer selected is the root layer.
+    if (_stage->GetRootLayer() != _stage->GetEditTarget().GetLayer()) {
+        TF_WARN(
```

Comment:
Seeing there are already two places with that code, please move it into either lib/mayaUsd/utils/restrictions.h (would be a new file) or lib/usdUfe/ufe/Utils.h (I really wish Utils.h were split into more specific files, but oh well... we got half a dozens of headers named utils.h all over the place.)